flink-core/src/main/java/org/apache/flink/api/java/typeutils/runtime/NullableSerializer.java

```diff
@@ -56,33 +59,51 @@
 public class NullableSerializer<T> extends TypeSerializer<T> {
 	private static final long serialVersionUID = 3335569358214720033L;
 
+	@Nonnull
 	private final TypeSerializer<T> originalSerializer;
 
-	private NullableSerializer(TypeSerializer<T> originalSerializer) {
+	private NullableSerializer(@Nonnull TypeSerializer<T> originalSerializer) {
 		Preconditions.checkNotNull(originalSerializer, "The original serializer cannot be null");
```

Comment:
You could also remove the `Preconditions` check now that the annotation is in place.